applications/GeoMechanicsApplication/tests/cpp_tests/custom_strategies/test_geomechanics_time_integration_scheme.cpp

```diff
@@ -194,21 +194,16 @@ KRATOS_TEST_CASE_IN_SUITE(FunctionCallsOnAllConditions_AreOnlyCalledForActiveCon
     tester.TestFunctionCallOnAllComponents_AreOnlyCalledForActiveComponents<SpyCondition>();
 }
 
-KRATOS_TEST_CASE_IN_SUITE(FunctionCalledOnCondition_IsCalledOnActiveAndInactiveConditions,
+KRATOS_TEST_CASE_IN_SUITE(FunctionCalledOnElement_IsCalledOnActiveAndInactiveElementsAndConditions,
                           KratosGeoMechanicsFastSuiteWithoutKernel)
-{
-    GeoMechanicsSchemeTester tester;
-    tester.TestFunctionCalledOnComponent_IsCalledOnActiveAndInactiveComponents<SpyCondition>(
-        [](auto& rModelPart, auto& rCondition) { rModelPart.AddCondition(rCondition); },
-        [](auto& rScheme, auto& rModelPart) { rScheme.InitializeConditions(rModelPart); });
-}
-
-KRATOS_TEST_CASE_IN_SUITE(FunctionCalledOnElement_IsCalledOnActiveAndInactiveElements, KratosGeoMechanicsFastSuiteWithoutKernel)
 {
     GeoMechanicsSchemeTester tester;
     tester.TestFunctionCalledOnComponent_IsCalledOnActiveAndInactiveComponents<SpyElement>(
         [](auto& rModelPart, auto& rElement) { rModelPart.AddElement(rElement); },
         [](auto& rScheme, auto& rModelPart) { rScheme.InitializeElements(rModelPart); });
+    tester.TestFunctionCalledOnComponent_IsCalledOnActiveAndInactiveComponents<SpyCondition>(
+        [](auto& rModelPart, auto& rCondition) { rModelPart.AddCondition(rCondition); },
+        [](auto& rScheme, auto& rModelPart) { rScheme.InitializeConditions(rModelPart); });
```

Comment:
From a testing point of view, I think it's better to keep these unit tests separated rather than merging them. I think we can achieve that by extending the lambda expression as follows:
```suggestion
        [](auto& rScheme, auto& rModelPart) {
            rScheme.SetElementsAreInitialized(); // precondition
            rScheme.InitializeConditions(rModelPart); });
```
This also makes the precondition (i.e. elements must have been initialized before conditions are initialized) explicit.

You could even consider to assert that the precondition is properly checked.